src/injected/web/requests.js

```diff
@@ -112,7 +119,59 @@ function start(req, id) {
 }
 
 function getFullUrl(url) {
-  const a = document.createElementNS(NS_HTML, 'a');
-  a.setAttribute('href', url);
-  return a.href;
+  const a = document::createElementNS(NS_HTML, 'a');
+  a::setAttribute('href', url);
+  return a::getHref();
+}
+
+const { keys, getAll } = FormData.prototype;
+const { FileReader } = global;
+const { readAsArrayBuffer } = FileReader.prototype;
+
+async function encodeBody(body) {
+  const cls = getType(body);
+  switch (cls) {
+  case 'formdata': {
+    const data = {};
+    const resolveKeyValues = async (key) => {
+      const values = body::getAll(key)::map(encodeBody);
+      data[key] = await Promise.all(values);
+    };
+    await Promise.all([...body::keys()]::map(resolveKeyValues));
```

Comment:
`map` is incorrect here.
I think the binding syntax `::` is too early to use because both tsserver and ESLint do not support it and thus cannot prevent such errors for us.

Reply:
Why is map incorrect here? I've checked the original code once again and I don't see the issue.

As for TypeScript, it's also managed by a big bureaucratic entity so they won't implement support for this syntax until it's at least Stage-3 or 4. Do you plan to rewrite everything in TypeScript though?

Reply:
```js
// somewhere above
const map = {};

// Oops, map is not a function
await Promise.all([...body::keys()]::map(resolveKeyValues));
```

Reply:
Geez, I'm blind. I've spotted `keys` a few times, but missed this one.